src/random.cpp

```diff
@@ -86,6 +86,13 @@ uint32_t zmq::generate_random ()
 //  order fiasco, this is done using function-local statics, if the
 //  compiler implementation supports thread-safe initialization of those.
 //  Otherwise, we fall back to global statics.
+//  HOWEVER, this initialisation code adds a race condition when an
```

Comment:
It is not a race condition (this has nothing to do with concurrency). Maybe:

```
HOWEVER, this initialisation code imposes ordering constraints, which are not obvious to users of libzmq, and may lead to problems if atexit or similar methods are used for cleanup.
```

Reply:
Thanks, updated